Sources/Build/BuildDescription/SwiftTargetBuildDescription.swift

```diff
@@ -291,6 +291,70 @@ public final class SwiftTargetBuildDescription {
         }
 
         try self.generateResourceEmbeddingCode()
+        try self.generateTestObservation()
+    }
+
+    private func generateTestObservation() throws {
+        guard target.type == .test, buildParameters.targetTriple.isDarwin(), buildParameters.experimentalTestOutput else {
+            return
+        }
+
+        // FIXME: Ensure we don't make multiple of these.
```

Comment:
This needs to be fixed before landing

Reply:
Interestingly, there doesn't seem to be a problem here in practice? I have been using this for a bit now when running SwiftPM's own tests and it seems to work fine.

Reply:
can we use some invocation identifier?

Reply:
I think we just want to generate the code once for the entire package since we build a single bundle.

Reply:
Fixed now.